// dynamic-form-json/dist/components/_fieldStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Label = exports.Upload = exports.TextArea = exports.Select = exports.Input = exports.FieldContainer = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const FieldContainer = _styledComponents.default.div`
  width: 100%;
  margin-bottom: 10px;
  .label {
    font-weight: bold;
  }
  .error {
    color: #ff0000;
  }

  @media screen and (min-width: 1024px) {
    width: 300px;
  }
`;
exports.FieldContainer = FieldContainer;
const Input = _styledComponents.default.input`
  width: 100%;
  margin: 5px 5px 4px 0;
  padding: 5px;
  box-sizing: border-box;

  @media screen and (min-width: 1024px) {
    width: 300px;
  }
`;
exports.Input = Input;
const Select = _styledComponents.default.select`
  width: 100%;
  min-width: 190px;
  margin: 5px 5px 4px 0;
  padding: 5px;
  box-sizing: border-box;

  @media screen and (min-width: 1024px) {
    width: 300px;
  }
`;
exports.Select = Select;
const TextArea = _styledComponents.default.textarea`
  width: 100%;
  min-width: 180px;
  box-sizing: border-box;
  padding: 5px;
  height: 75px;

  @media screen and (min-width: 1024px) {
    width: 300px;
  }
`;
exports.TextArea = TextArea;
const Upload = _styledComponents.default.input`
  width: 100%;
  min-width: 180px;
  box-sizing: border-box;
  padding: 5px;
  height: 30px;

  @media screen and (min-width: 1024px) {
    width: 300px;
  }
`;
exports.Upload = Upload;
const Label = _styledComponents.default.label`
  margin-right: 20px;
`;
exports.Label = Label;
