// dynamic-form-json/dist/components/RadioButtonField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _fieldStyles = require("./_fieldStyles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function RadioButtonField(props) {
  return /*#__PURE__*/_react.default.createElement(_fieldStyles.FieldContainer, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "label"
  }, props.label), props.options.map((opt, index) => {
    return /*#__PURE__*/_react.default.createElement(_fieldStyles.Label, {
      key: index
    }, /*#__PURE__*/_react.default.createElement("input", {
      type: "radio",
      name: props.name,
      value: opt,
      checked: opt === props.value,
      onBlur: props.handleBlur,
      onChange: props.onChange
    }), opt);
  }), props.error && props.touched[props.name] && /*#__PURE__*/_react.default.createElement("div", {
    className: "error"
  }, props.error));
}

RadioButtonField.propTypes = {
  label: _propTypes.default.string,
  name: _propTypes.default.string.isRequired,
  value: _propTypes.default.any,
  options: _propTypes.default.array,
  error: _propTypes.default.any,
  onChange: _propTypes.default.func.isRequired
};
var _default = RadioButtonField;
exports.default = _default;
